pkg/config/node.go

```diff
@@ -521,7 +533,7 @@ func GetConfig(kubeconfigPath, clusterConfigPath, resolvConfPath string, apiVips
 	return nodes[0], nil
 }
 
-func getNodeConfig(kubeconfigPath, clusterConfigPath, resolvConfPath string, apiVip net.IP, ingressVip net.IP, apiPort, lbPort, statPort uint16) (node Node, err error) {
+func getNodeConfig(kubeconfigPath, clusterConfigPath, resolvConfPath string, apiVip net.IP, ingressVip net.IP, apiPort, lbPort, statPort uint16, onPremPlatform bool) (node Node, err error) {
```

Comment:
If possible, let's not rely on the computed state and passed value of `onPremPlatform bool`, because by not handling the error from `if onPremPlatform, err := isOnPremPlatform(clusterConfigPath); !onPremPlatform {` we will never notice if someone called `getNodeConfig` without thinking if they correctly pass `true` or `false`.

Calling `isOnPremPlatform` inside `getNodeConfig` is not expensive so I'd rather keep it there. If in the future, for any reason we realize there is a performance bottleneck inside this function, then we will handle it

Reply:
Done.